### PR TITLE
goal: resolve data dir code duplication

### PR DESCRIPTION
**Pull request by abebeos**
re #5068

leaving it unsquashed for easier review.

(~note the first commits took multiple hours, due to usual trial&error of solution paths. Please intervene only if something is very wrong. Details can be changed once its finalized, like e.g. keeping possibly `dataDirs` encapsulated.~)

### CODE REVIEW

File: cmd/util/datadir/report.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package datadir
+
+// TODO: eliminate duplication (copied from cmd/algocfg/report.go)
```
Comment (Eric-Warehime) [inline cmd/util/datadir/report.go:19]: I personally prefer TODOs to be captured in issues and not inline, but I've seen other people's commits w/ them left in as well so I don't think there's a consensus on this.

**Review by Eric-Warehime — APPROVED** [review]
LGTM

File: cmd/util/datadir/report.go (hunk 1 above)
Comment (abebeos) [reply cmd/util/datadir/report.go:19]: > I personally prefer TODOs to be captured in issues

Agree,. This one id mostly a marker for myself (immediate next step after merge).

**Review by winder — APPROVED** [review]

**Comment by winder** [comment]
Thanks for the contribution!

**Pull request merged**